### PR TITLE
[Bug fix] asinh_bw: keep the gradient finite once x*x overflows

### LINKED ISSUE #55725: [Bug] ttnn.asinh_bw returns a zero gradient for |x| > 2^64, 24% of the bfloat16 domain: 1 + x*x overflows before the rsqrt
### Describe the bug

`ttnn.asinh_bw` returns a gradient of exactly `0` for every `|x| > 2^64`, which is **15,874 of the 65,536 bfloat16 bit patterns — 24% of the domain** — although `d/dx asinh(x) = 1/sqrt(1 + x^2)` tends to `1/|x|` and stays a normal float for every representable input up to `2^126`.

The composite squares the input before the reciprocal square root
([`unary_backward.cpp#L1043-L1058`](https://github.com/tenstorrent/tt-metal/blob/main/ttnn/cpp/ttnn/operations/eltwise/unary_backward/unary_backward.cpp#L1043-L1058)):

```cpp
std::vector<EltwiseUnaryWithParam> ops_chain = {
    EltwiseUnaryWithParam{UnaryOpType::SQUARE},
    EltwiseUnaryWithParam{UnaryOpType::ADD_UNARY_SFPU, 1.0f},
    EltwiseUnaryWithParam{UnaryOpType::RSQRT}};
Tensor grad_result =
    ttnn::multiply(grad, ttnn::unary_chain(input, ops_chain, output_mem_config), std::nullopt, output_mem_config);
```

`x * x` overflows to `+inf` for `|x| > sqrt(FLT_MAX) = 1.8446743e19`, `inf + 1` is `inf`, and `rsqrt(inf)` is `0`. The true derivative at the smallest overflowing input, `x = 2^64`, is `5.42e-20` — nowhere near the bottom of the float range — and it is still `1.18e-38`, a normal float, at `x = 2^126`.

The same overflow is in `acosh_bw` (`rsqrt(x*x - 1)`), and in a much narrower form in `atan_bw` / `atanh_bw` (`1/(1 +/- x^2)`, where the true result is already subnormal past `2^63`, so only 2 bit patterns are affected). `acosh_bw` is being rewritten by #53885, so this report does not propose touching it.

### Repro

```python
import torch, ttnn

device = ttnn.open_device(device_id=0)
x = torch.tensor([[1.0, 1e10, 1.8446744e19, 3.6893488e19, 1e25, 1e30, 8.507059e37]], dtype=torch.float32)
t = ttnn.from_torch(x, dtype=ttnn.float32, layout=ttnn.TILE_LAYOUT, device=device)
g = ttnn.from_torch(torch.ones_like(x), dtype=ttnn.float32, layout=ttnn.TILE_LAYOUT, device=device)

print(ttnn.to_torch(ttnn.asinh_bw(g, t)[0]))
# tensor([[7.0711e-01, 1.0000e-10, 0., 0., 0., 0., 0.]])
xd = x.double()
print(1.0 / torch.sqrt(1.0 + xd * xd))
# tensor([[7.0711e-01, 1.0000e-10, 5.4210e-20, 2.7105e-20, 1.0000e-25, 1.0000e-30, 1.1755e-38]])
```

### Observed vs expected

float32, gradient `1`, golden `1/sqrt(1 + x^2)` in float64. ULP is the integer distance between float32 bit patterns:

| input | got | want | ULP |
|---|---|---|---|
| 1 | 0.7071068 | 0.7071068 | 0 |
| 1e10 | 9.999999e-11 | 1e-10 | 1 |
| 1.8446744e19 (2^64) | **0** | 5.421011e-20 | 528,482,304 |
| 3.6893488e19 (2^65) | **0** | 2.710505e-20 | 520,093,696 |
| 1e25 | **0** | 1e-25 | 368,547,464 |
| 1e30 | **0** | 1e-30 | 228,737,632 |
| 8.507059e37 (2^126) | **0** | 1.175494e-38 | 8,388,608 |

bfloat16, gradient `1`, ULP between bfloat16 bit patterns:

| input | got | want | ULP |
|---|---|---|---|
| 1.8446744e19 (2^64) | **0** | 5.421011e-20 | 8,064 |
| 3.6893488e19 (2^65) | **0** | 2.710505e-20 | 7,936 |
| 9.97364e24 | **0** | 1.001664e-25 | 5,624 |
| 1.00026e30 | **0** | 9.984021e-31 | 3,490 |
| 8.507059e37 (2^126) | **0** | 1.175494e-38 | 128 |

The relative error over the whole band is 100%.

### Affected input range

All 65,536 bfloat16 bit patterns as the input, gradient `1`, counting inputs whose exact gradient is a normal float but whose result is exactly zero (subnormal inputs excluded):

| dtype | zeros where the golden is normal | of the domain | affected range | max ULP | mean ULP |
|---|---|---|---|---|---|
| bfloat16 | 15,874 | 24.2% | `2^64 <= \|x\| <= 8.51e37` | 8,064 | 996.38 |
| float32 | 15,874 | 24.2% | `2^64 <= \|x\| <= 8.51e37` | 528,482,304 | 65,295,046 |

### Which architectures

All of them. `asinh_bw` is a host-side composite in `ttnn/cpp/ttnn/operations/eltwise/unary_backward/unary_backward.cpp`, so Wormhole and Blackhole build the same op graph and produce the same values.

### How it was verified

ttsim Blackhole v1.10.6 (`TT_METAL_SIMULATOR=libttsim_bh.so`, slow dispatch). Exhaustive sweep of all 65,536 bfloat16 bit patterns as the input with an all-ones gradient, in both bfloat16 and float32, against `1/sqrt(1 + x^2)` evaluated in float64. Subnormal inputs (flushed by the hardware) are excluded.

`test_backward_asinh.py::test_bw_asinh` does not catch it: it draws inputs from `[-100, 100]`, twenty orders of magnitude below the overflow.

asinh_bw before/after

### PR DESCRIPTION
**Pull request by badnikhil**
### PR Category

Bug fix

### Summary

Closes #55725

`asinh_bw` evaluates `1/sqrt(1 + x*x)` as a `SQUARE -> ADD 1 -> RSQRT` chain. `x * x` overflows to `+inf` for `|x| > sqrt(FLT_MAX)`, and `rsqrt(inf)` is `0`, so the gradient vanished over the whole `2^64 < |x| <= FLT_MAX` band — 15,874 of the 65,536 bfloat16 bit patterns, 24% of the domain — even though `1/sqrt(1 + x^2)` tends to `1/|x|` and is still `1.18e-38`, a normal float, at `x = 2^126`.

Past `2^62` the two agree: `1/x^2` is below `2^-124` there, so `1/sqrt(1 + x^2)` and `1/|x|` differ by far under half a float32 ULP. The chain is kept for `|x| <= 2^62` and the reciprocal is taken above it, so nothing that was already correct moves.

All 65,536 bfloat16 bit patterns as the input with an all-ones gradient, golden `1/sqrt(1 + x^2)` in float64, on ttsim Blackhole:

| dtype | zeros where the golden is normal | max ULP | mean ULP |
|---|---|---|---|
| bfloat16 | 15,874 → **2** | 8,064 → **128** | 996.3788 → **0.0639** |
| float32 | 15,874 → **2** | 528,482,304 → **8,388,608** | 65,295,045.59 → **1,221.687** |

Spot values, float32:

| input | before | after | exact |
|---|---|---|---|
| 1 | 0.7071068 | 0.7071068 | 0.7071068 |
| 1e10 | 9.999999e-11 | 9.999999e-11 | 1e-10 |
| 2^64 | 0 | **5.421011e-20** | 5.421011e-20 |
| 1e25 | 0 | **1e-25** | 1e-25 |
| 1e30 | 0 | **1e-30** | 1e-30 |
| 2^126 | 0 | 0 | 1.175494e-38 |

asinh_bw before/after

### Notes for reviewers

- Host composite, so both Wormhole and Blackhole are covered by the one change.
- The two residual zeros and the residual ULP are `ttnn.reciprocal`'s own profile at the bottom of the normal range, reproduced exactly: sweeping all 65,536 bit patterns through `ttnn.reciprocal` alone gives the same 2 zeros (at `x = ±2^126`, where `1/x` is `FLT_MIN` and the op returns `0`), the same max ULP of 8,388,608, and mean 1,221.70 against this PR's 1,221.687. Nothing in this change makes them worse, and fixing them belongs in `reciprocal`.
- Cutoff is `2^62`, not the `2^64` where the overflow actually starts, so the branch is not sitting on the boundary. Over `2^62 < |x| <= 2^64` both expressions are correct and the reciprocal is chosen; the relative difference there is at most `2^-125`.
- Costs three extra dispatches (`abs`, `gt`, a two-step `ABS -> RECIP` chain) plus the `where`. A branch is unavoidable: float32 has no scale factor that keeps `x^2` finite for `|x| = FLT_MAX` while keeping `1` representable, since `2^-64 * 2^-64` is subnormal.
- Special values are unchanged. `|±inf| > 2^62` selects `1/inf = 0`, which is the correct limit; NaN fails the ordered `gt` and takes the chain, which yields NaN as before; `±0` takes the chain and yields `1`.
- `acosh_bw` (`rsqrt(x*x - 1)`) and, much more narrowly, `atan_bw` / `atanh_bw` (`1/(1 ± x^2)`, only 2 bit patterns because the true result is subnormal past `2^63`) carry the same overflow. `acosh_bw` is being rewritten by #53885, so this PR deliberately leaves all three alone rather than conflict with it.
- `test_bw_asinh_all_bitpatterns` sweeps every bfloat16 bit pattern in both bfloat16 and float32, asserts no gradient above `2^-100` comes back as zero, and holds 4 ULP over that range. It fails both parametrisations on `main` ("9218 inputs returned a zero gradient, first at x=1.8446744073709552e+19"). The `2^-100` floor keeps the assertion off the `ttnn.reciprocal` boundary described above; the reference is evaluated in float64 rather than through the registered golden.
- The pre-existing `test_bw_asinh` PCC cases still pass; they did not catch this because they draw inputs from `[-100, 100]`, twenty orders of magnitude below the overflow.
- Verified on ttsim Blackhole v1.10.6 (slow dispatch); no silicon.

### CODE REVIEW

**Review by mateusznowakTT — CHANGES_REQUESTED** [review]
Approach looks good improving the op accuracy.
Please rewrite it to use new unary backward infra available since https://github.com/tenstorrent/tt-metal/pull/56061 .
Please make sure to address CI failures.
